### PR TITLE
Add pyright

### PR DESCRIPTION
**Pull request by oskarrrrrrr**
I added [pyright](https://github.com/microsoft/pyright) and fixed some of the errors that it produced. Mostly unused imports. Pyright will run in `workflow/tests.yml`.

Added a Makefile so that it's trivial to run tests, linters etc. I didn't add e2e test there yet.

I used poetry 1.3.x, if that's not ok for any reason I can change it.

### CODE REVIEW

**Comment by raven-wing** [comment]
Cool idea with makes, as for e2e tests I'd leave it as it is for now. We'll probably want to move them to `goodmap-integration` and to rewrite some tests.

**Comment by oskarrrrrrr** [comment]
I resolved the conflicts and disabled some pyright errors. Pyright can be a bit pesky if you're not used to it. You can always add a comment `# pyright: ignore` or ask me for help.

**Comment by raven-wing** [comment]
> I resolved the conflicts and disabled some pyright errors. Pyright can be a bit pesky if you're not used to it. You can always add a comment `# pyright: ignore` or ask me for help.

After merge, it will be part of merge blocker, so it will not be a problem in future. Yet still waiting for one of comments to be resolved to close this issue.

**Comment by raven-wing** [comment]
> > I resolved the conflicts and disabled some pyright errors. Pyright can be a bit pesky if you're not used to it. You can always add a comment `# pyright: ignore` or ask me for help.
> 
> After merge, it will be part of merge blocker, so it will not be a problem in future. Yet still waiting for one of comments to be resolved to close this issue.

My bad... Sorry for waiting... I did add comment yet didn't publish the review. :(

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+install:
+	python -m pip install --upgrade pip
+	pip install poetry
+	poetry install
```
[the remark this answers is not in the document]
Comment (oskarrrrrrr) [reply Makefile]: > For dealing with system-wide requirements in projects Makefile I'd say no.
It's kind of on the user to use pyenv, venv or whatever they need/want.

I went with 2. as it's easier.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -22,14 +22,11 @@ jobs:
...
     - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install poetry
-        poetry install
-    - name: Test with pytest
-      run: |
-        poetry run coverage run -m pytest
-        poetry run coverage lcov
+      run: make install
```
Comment (raven-wing) [inline .github/workflows/tests.yml:25]: It seems this is no longer available

**Review by raven-wing — APPROVED** [review]

**Pull request merged**